runtime/src/main/java/com/dylibso/chicory/runtime/alloc/DefaultMemAllocStrategy.java

```diff
@@ -0,0 +1,23 @@
+package com.dylibso.chicory.runtime.alloc;
+
+public final class DefaultMemAllocStrategy implements MemAllocStrategy {
+    private final int max;
+
+    public DefaultMemAllocStrategy(int max) {
+        this.max = max;
+    }
+
+    @Override
+    public int initial(int min) {
+        return min;
+    }
+
+    @Override
+    public int next(int current, int target) {
+        int next = (current <= 0) ? target : current;
+        while (next < target && next < max) {
+            next = next << 1;
```

Comment:
Can you add a unit test for this class?

I think this will overflow when `current` is greater than `Integer.MAX_VALUE / 2` and return a negative value

Reply:
Added the unit test, it doesn't break, please take a look and let me know if I missed any case.